### ISSUE TITLE
Front end JS errors in console after Recent Update

### ISSUE THREAD

**Issue by TGoodrich** [getk2/k2#284]
Saw these errors on my live site after recent k2 dev update. I checked my staging site which I had not updated yet and there was no errors.

I just now updated k2 DEV v2.7 on staging site and am now getting the same JS errors in the front end as I am on the live site.

I'm not sure what is causing the errors but I can confirm it is caused by K2.

I included 2 snapshots, 1 before k2 update and 1 after updating k2. Errors shown in image
![site-no-errors](https://cloud.githubusercontent.com/assets/10102765/13712960/7c21334e-e794-11e5-90b6-cfe9b5d9abd8.gif)
![site-with-errors](https://cloud.githubusercontent.com/assets/10102765/13712964/83567476-e794-11e5-9e2f-9d77949a9573.gif)

**Comment by fevangelou** [comment]
We have dropped Mootools support in K2 v2.7.0. Maybe this has simply exposed these errors. If you look closely, you'll see that these files are not related to K2.

Are there any visual differences besides these JS warnings?

**Comment by TGoodrich** [comment]
Did you drop Mootools support for all past versions of K2 v2.7.0. as well? I only ask because I was able to find a previous K2 v2.7.0. build in which I downloaded this past January. After installing the old build over the most recent one the JS errors went away.

To re-verify I then installed the most recent K2 v2.7.0. again and the errors came back. So even though these files are not related to K2, K2 is somehow causing these errors. I do not know what has changed in the latest build but the old builds do not cause these errors.

**Comment by TGoodrich** [comment]
If it helps the only difference I can see is that the js script ordering has changed in the page head. I'm guessing this is now why the errors are occurring. I do not really understand what mechanism determines script order but it seems something has changed in the latest K2 v2.7.0. that has thrown off the ordering.

If you would like I could post a couple of screenshots showing the ordering. 1 screenshot of my live site running old build and 1 on my staging site running new build, both showing the js script order.

Both sites are identical except for the version of K2 v2.7.0.

**Comment by TGoodrich** [comment]
Ok after more digging I found that after disabling BT SmartSearch module the errors go away. Still not sure however why these errors did not occur with older  K2 v2.7.0. builds.

**Comment by fevangelou** [comment]
The order of how JS files are loaded is determined by the placement of the K2 system plugin. Your test site may have a different ordering for that plugin. The absence of Mootools support combined with a changed ordering for the K2 system plugin is the reason why you saw these warnings I guess.

If you wish to make sure K2's CSS and JS are ordered last in the <head> of the HTML document, simply re-order the plugin to be as nearly at the bottom of the system plugins list. I say "near" because you should keep any caching plugins at the very bottom of that list.

Verify and let me know.

**Comment by fevangelou** [comment]
Oh and BTW, K2 has a very nice search module with auto-suggest, built into the module K2 Tools :)

**Comment by TGoodrich** [comment]
Thanks for the tip on how the JS files are loaded. I've been using Joomla for 3 years or so now and I never knew this. Now that I do, I'm sure it will save me from some headaches in the future :) I'm not going to mess around with the plugin ordering as of now though because my issue was solved by disabling the BT SmartSearch module.

Also, thanks for suggesting the use of the K2 Tools module - for Search. I do not know why I didn't try it before. Since your suggestion I have implemented it on my testing site and I really like it.. especially the live search. Once I'm done with customizing the module I plan on implementing it on my live site.

Cheers :)
